packages/react-core/src/components/HelperText/__tests__/HelperText.test.tsx

```diff
@@ -1,66 +1,75 @@
+  const { asFragment } = render(
+    <HelperText>
+      <HelperTextItem>help test text</HelperTextItem>
+      <HelperTextItem>help test text 2</HelperTextItem>
+      <HelperTextItem>help test text 3</HelperTextItem>
+    </HelperText>
+  );
+  expect(asFragment()).toMatchSnapshot();
+});
 
-  test('variant comonent helper text renders properly', () => {
-    const { asFragment } = render(
-      <HelperText component="ul">
-        <HelperTextItem component="li">help test text 1</HelperTextItem>
-        <HelperTextItem component="li">help test text 2</HelperTextItem>
-      </HelperText>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('HelperText renders custom className', () => {
+  render(
+    <HelperText className="custom" data-testid="helper">
+      <HelperTextItem>help test text 1</HelperTextItem>
+    </HelperText>
+  );
+  expect(screen.getByTestId('helper')).toHaveClass('custom');
+});
 
-  test('icon helper text renders properly', () => {
-    const { asFragment } = render(
-      <HelperText>
-        <HelperTextItem icon={<CheckIcon />}>help test text</HelperTextItem>
-      </HelperText>
+Object.values(['indeterminate', 'warning', 'success', 'error']).forEach((variant) => {
+  test(`Renders ${styles.modifiers[variant]} when variant = ${variant}`, () => {
+    render(
+      <HelperTextItem variant={variant as 'default' | 'indeterminate' | 'warning' | 'success' | 'error'}>
+        text
+      </HelperTextItem>
     );
-    expect(asFragment()).toMatchSnapshot();
+    expect(screen.getByText('text').parentElement).toHaveClass(styles.modifiers[variant]);
   });
+});
 
-  test('dynamic helper text renders successfully', () => {
-    const { asFragment } = render(
-      <HelperText>
-        <HelperTextItem isDynamic>help test text</HelperTextItem>
-      </HelperText>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('HelperTextItem renders custom className', () => {
+  render(
+    <HelperText>
+      <HelperTextItem className="custom">help test text 1</HelperTextItem>
+    </HelperText>
+  );
+  expect(screen.getByText('help test text 1').parentElement).toHaveClass('custom');
+});
 
-  test('helper text block renders successfully', () => {
-    const { asFragment } = render(
-      <HelperText>
-        <HelperTextItem>help test text 1</HelperTextItem>
-        <HelperTextItem>help test text 2</HelperTextItem>
-        <HelperTextItem>help test text 3</HelperTextItem>
-      </HelperText>
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Renders as ul when component = ul', () => {
+  render(
+    <HelperText component="ul" aria-label="helper">
+      <HelperTextItem component="li">help test text 1</HelperTextItem>
+      <HelperTextItem component="li">help test text 2</HelperTextItem>
+    </HelperText>
+  );
+  expect(screen.getByLabelText('helper').tagName).toBe('UL');
+  expect(screen.getByText('help test text 1').parentElement?.tagName).toBe('LI');
```

Comment:
Let's split this up into separate tests. This test should check that the `ul` tag is there, then for aria-label we can have tests for:

- checking the aria-label isn't rendered if the component is not ul
- checking the aria-label does get rendered when component = ul

We should also add a couple of tests that check the `role` attribute doesn't get applied when component is not ul, and that it does get applied when component = ul